File: server/src/main/java/com/kennethrdzg/socialapp/service/PostService.java
package com.kennethrdzg.socialapp.service;

import java.util.List;

import com.kennethrdzg.socialapp.entities.Post;

public interface PostService {
    public Post uploadPost(Post post);
    public List<Post> getPosts();
    public List<Post> getPosts(int page);
    public Post getPostById(int postId);
    public List<Post> getPostsByUserId(int userId);
}

File: smalltalk-client/src/app/services/post.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Post } from '../entities/post';
import { UserToken } from '../entities/user-token';

@Injectable({
  providedIn: 'root'
})
export class PostService {
  private apiUrl: string = 'http://localhost:8080';

  constructor(private http: HttpClient) {}

  getPosts(userToken: UserToken): Observable<Array<Post>>{
    return this.http.get<Array<Post>>(this.apiUrl + "/posts/"+userToken.userId+"/"+userToken.token);
  }

  uploadPost(userToken: UserToken, content: string): Observable<Post>{
    return this.http.post<Post>(this.apiUrl + "/posts/upload", {"username": userToken.username, "token": userToken.token, "content": content});
  }

  likeButtonPressed(userToken: UserToken, postId: number, liked: boolean): Observable<Post>{
    return this.http.post<Post>(this.apiUrl + "/posts/like", {"username": userToken.username, "token": userToken.token, "id": postId, "liked": liked});
  }
}
